pyproject.toml

```diff
@@ -84,3 +79,46 @@ sphinxcontrib-mermaid = "^0.9.2"
 [build-system]
 build-backend = "poetry.core.masonry.api"
 requires = ["poetry-core>=1.2.0"]
+
+[tool.pytest.ini_options]
+addopts = """	
+  -v
+  -p no:warnings
+  --cov=virtual_ecosystem
+  --cov-report=html:htmlcov/coverage
+  --doctest-modules --ignore=virtual_ecosystem/__main__.py
+  """
+python_files = 'test_*.py'
+testpaths = ['tests']
+
+[tool.ruff]
+target-version = "py310"
+
+[tool.ruff.lint]
+# Enable all `pydocstyle` rules, limiting to those that adhere to the
+# Google convention via `convention = "google"`, below.
+select = [
+  # "B", # flake8-bugbear
+  # "SIM", # flake8-simplify
+  "E", # pycodestyle
+  "F", # pyflakes
+  "D", # pydocstyle
+  "I", # isort
+  "UP", # pyupgrade
```

Comment:
I also quite like the `"RUF"` checks, which are a few `ruff`-specific ones.

Reply:
```
Found 97 errors.
```

But they all seem sensible, so will adopt it.